apps/interop-rxjs-signal/src/app/list/photos.store.ts

```diff
@@ -30,106 +36,90 @@ const initialState: PhotoState = {
   error: '',
 };
 
-@Injectable()
-export class PhotoStore
-  extends ComponentStore<PhotoState>
-  implements OnStoreInit, OnStateInit
-{
-  private photoService = inject(PhotoService);
+const getSavedState = (): PhotoState => {
+  const savedJSONState = localStorage.getItem(PHOTO_STATE_KEY);
+  const parsedState = savedJSONState && JSON.parse(savedJSONState);
+  return parsedState
+    ? { ...defaultState, search: parsedState.search, page: parsedState.page }
+    : defaultState;
+};
 
-  private readonly photos$ = this.select((s) => s.photos);
-  private readonly search$ = this.select((s) => s.search);
-  private readonly page$ = this.select((s) => s.page);
-  private readonly pages$ = this.select((s) => s.pages);
-  private readonly error$ = this.select((s) => s.error);
-  private readonly loading$ = this.select((s) => s.loading);
+@Injectable({ providedIn: 'root' })
+export class PhotoStore {
+  private readonly photoService = inject(PhotoService);
 
-  private readonly endOfPage$ = this.select(
-    this.page$,
-    this.pages$,
-    (page, pages) => page === pages
-  );
+  // Signals
+  readonly state = signal(getSavedState());
+  readonly photos = computed(() => this.state().photos);
+  readonly page = computed(() => this.state().page);
+  readonly pages = computed(() => this.state().pages);
+  readonly error = computed(() => this.state().error);
+  readonly loading = computed(() => this.state().loading);
+  readonly endOfPage = computed(() => this.page() === this.pages());
+
+  // Sources
+  readonly searchControl = new FormControl({
+    value: getSavedState().search,
+    disabled: false,
+  });
 
-  readonly vm$ = this.select(
-    {
-      photos: this.photos$,
-      search: this.search$,
-      page: this.page$,
-      pages: this.pages$,
-      endOfPage: this.endOfPage$,
-      loading: this.loading$,
-      error: this.error$,
-    },
-    { debounce: true }
+  readonly search$ = this.searchControl.valueChanges.pipe(
+    startWith(this.searchControl.value),
+    filter((a) => !!a && a.length >= 3),
```

Comment:
you can write `filter((a): a is string => !!a && a.length >= 3)` to avoid having string | null